guava/src/main/java/com/fasterxml/jackson/datatype/guava/ser/GuavaBeanSerializerModifier.java

```diff
@@ -24,7 +28,13 @@ public List<BeanPropertyWriter> changeProperties(SerializationConfig config,
     {
         for (int i = 0; i < beanProperties.size(); ++i) {
             final BeanPropertyWriter writer = beanProperties.get(i);
-            if (Optional.class.isAssignableFrom(writer.getType().getRawClass())) {
+            // [databind#0000] Since 2.17 : problem to solve + reason for matching Unwrapping....Writer
+            if (writer instanceof UnwrappingBeanPropertyWriter) {
```

Comment:
Oh actually... shouldn't this only apply to `Optional` type properties? I don't think we want to change standard UnwrappingBeanPropertyWriter for other types, otherwise it'd change behavior

So, I think this if statement should be inside `if (Optional.class.isAssignableFrom(...))`, not alternative

Reply:
Ah, excellent catch. I believe my initial PR was accurate. The error occurred when I attempted to rewrite it from memory the second time around. The issue has now been resolved. Thanks :)